src/App.js

```diff
@@ -1,29 +1,60 @@
 import React, { useEffect, useState } from "react";
 import fetchCountries from "./api/fetchCountries";
+import fetchCityImage from "./api/fetchCityImage";
+import fetchWeather from "./api/fetchWeather";
+import { GridLoader } from "react-spinners";
 
 function App(response) {
+  const [loading, setLoading] = useState(false);
+
   const [countries, setCountries] = useState();
   const [country, setCountry] = useState("United Kingdom");
-  const [capital, setcapital] = useState("London");
+  const [capital, setCapital] = useState("London");
+  const [cityImage, setCityImage] = useState("London");
+  const [weather, setWeather] = useState("raining");
+
+  const weatherDesc = weather.current?.condition.text;
 
   useEffect(() => {
+    setLoading(true);
     fetchCountries().then((res) => {
       setCountries(res);
+      setLoading(false);
     });
   }, []);
 
+  useEffect(() => {
+    setLoading(true);
+    fetchWeather(capital).then((res) => {
+      setWeather(res);
+      fetchCityImage(capital).then((res) => {
+        setCityImage(res);
+        setLoading(false);
+      });
+    });
+  }, [capital]);
```

Comment:
This could cause 2 loading conditions to overwrite each other. Whichever fetch fn resolves first will set loading false, but the other could still be loading.

Probably not worth fixing it at this point since we are planning to use react-query soon!